pyro/infer/svgd.py

```diff
@@ -101,6 +103,81 @@ def log_kernel_and_grad(self, particles):
     def bandwidth_factor(self):
         return self._bandwidth_factor
 
+    @bandwidth_factor.setter
+    def bandwidth_factor(self, bandwidth_factor):
+        """
+        :param float bandwidth_factor: Optional factor by which to scale the bandwidth
+        """
+        if bandwidth_factor is not None:
+            assert bandwidth_factor > 0.0, "bandwidth_factor must be positive."
+        self._bandwidth_factor = bandwidth_factor
+
+
+@copy_docs_from(SteinKernel)
+class IMQSteinKernel(SteinKernel):
+    r"""
+    An IMQ (inverse multi-quadratic) kernel for use in the SVGD inference algorithm [1]. The bandwidth of the kernel
+    is chosen from the particles using a simple heuristic as in reference [2]. The kernel takes the form
+
+    :math:`K(x, y) = (\alpha + ||x-y||^2/h)^{\beta}`
+
+    where :math:`\alpha` and :math:`\beta` are user-specified parameters and :math:`h` is the bandwidth.
+
+    :param float alpha: Kernel hyperparameter, defaults to 0.5.
+    :param float beta: Kernel hyperparameter, defaults to -0.5.
+    :param float bandwidth_factor: Optional factor by which to scale the bandwidth, defaults to 1.0.
+    :ivar float bandwidth_factor: Property that controls the factor by which to scale the bandwidth
+        at each iteration.
+
+    References
+
+    [1] "Stein Points," Wilson Ye Chen, Lester Mackey, Jackson Gorham, Francois-Xavier Briol, Chris. J. Oates.
+    [2] "Stein Variational Gradient Descent: A General Purpose Bayesian Inference Algorithm," Qiang Liu, Dilin Wang
+    """
+    def __init__(self, alpha=0.5, beta=-0.5, bandwidth_factor=None):
+        """
+        :param float alpha: Kernel hyperparameter, defaults to 0.5.
+        :param float beta: Kernel hyperparameter, defaults to -0.5.
+        :param float bandwidth_factor: Optional factor by which to scale the bandwidth
+        """
+        assert alpha > 0.0, "alpha must be positive."
+        assert beta < 0.0, "beta must be negative."
+        if bandwidth_factor is not None:
```

Comment:
I would move this check into the setter, as you've done in RBFKernel.

Reply:
ah thanks this setter stuff always confuses me